libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp

```diff
@@ -287,7 +287,7 @@ bool NavEKF2_core::resetHeightDatum(void)
     // adjust the height of the EKF origin so that the origin plus baro height before and after the reset is the same
 
     if (validOrigin) {
-        if (!gpsGoodToAlign) {
```

Comment:
Why not fix the gpsGoodTo Align calculation instead? It clearly isn't if the GPS doesn't have a 3D fix and can also check other attributes such as reported vertical position accuracy.

Reply:
I'll have a look at the user log and come up with a recommendation for changes to calcGpsGoodToAlign(void) and also check where we are using the latest GPS as opposed to the buffered value as that could create a race condition given calcGpsGoodToAlign(void) is using latest.

Reply:
The race condition that can occur can't be eliminated due to the external calls to resetHeightDatum. On that basis I think this change  should go in but with the caveat that the 3D lock check is anded with gpsGoodToAlign.

Reply:
My concern is that using 3D fix lock as the sole check could result in an inaccurate height being baked in if the receiver is still recovering from a glitch.

Reply:
I've now removed the EKF change as discussed